modules/raft/src/main/java/org/apache/ignite/internal/raft/storage/segstore/IndexFileManager.java

```diff
@@ -120,20 +129,36 @@ class IndexFileManager {
         Files.createDirectories(indexFilesDir);
     }
 
+    void start() throws IOException {
+        try (Stream<Path> indexFiles = Files.list(indexFilesDir)) {
```

Comment:
Please delete all `tmp` files preemptively, I believe `start` is a right place for it. We should also have a test for restart with an incomplete index file in the folder.